### ISSUE TITLE
Java highlighter incorrectly highlights "enhanced for" loop if there is no space before the colon

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#672]
(Original issue 965 created by dreiss on 2014-03-13T16:53:20.411021+00:00)

See http://pygments.org/demo/290625/

In the text

```
List<String> strings = getStrings();
fail:
for (String aString: strings) {
  if (aString.equals("fail")) break fail;
  System.out.println(aString);
}
```

The variable aString appears to be highlighted as a label, but it's not a label. It's a variable name followed by the ":" in the java for loop syntax.

Interestingly, Bitbucket (which apparenly uses pygments) gets this right, but doesn't highlight labels at all if I don't specify the language as Java.  Compare:

```
#!java
List<String> strings = getStrings();
fail:
for (String aString: strings) {
  if (aString.equals("fail")) break fail;
  System.out.println(aString);
}
```

**Comment by Anteru** [comment]
(Original issue 965 last updated on 2014-10-07T20:37:45.328402+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-07T15:51:16.413821+00:00)

It is impossible to distinguish these cases without a more intricate rewrite of the lexer, which keeps track of things like "are we within a for statement head".

This is probably best done by someone who knows Java.

**Comment by Anteru** [comment]
(Original comment by tshatch on 2014-10-07T20:37:45.318133+00:00)

Fix highlighting of Java enhanced for loops.

Fixes #965

→ <<cset 1ed00bd80758>>
